simtools/corsika/corsika_config.py

```diff
@@ -52,7 +52,8 @@ def __init__(self, array_model, args_dict, db_config=None, label=None):
         self.azimuth_angle = None
         self._run_number = None
         self.config_file_path = None
-        self.primary_particle = self._set_primary_particle(args_dict)
+        # The following uses the setter defined below, that is why the args_dict is passed
+        self.primary_particle = args_dict
```

Comment:
This is correct but is without reading the functions below very confusing. Not nice that an additional comment is needed.

Reply:
What do you suggest we do then? The other option I see to make it more readable is to change to
```
self.primary_particle = {"primary_id_type": args_dict.get("primary_id_type"), "primary": args_dict.get("primary")}
```

But I think this is worse than the comment explaining it to future developers.

Reply:
No suggestion - I was just complaining. Should have made it clearer that no changes are expected.